### PR TITLE
feat(models): MiMoV2MTPForCausalLM (V2.5-Pro NextN/MTP draft model)

### PR DESCRIPTION
**Pull request by zorrofox**
Part of #1053 (P1-3).

## What

Single-layer draft model for MiMo-V2.5-Pro 3-layer MTP. One instance loads exactly one layer from `model_mtp.safetensors` (selected via `config.mtp_layer_idx`, default 0). The `MultiLayerDraftWorker` (#1053 P1-4) will create one instance per layer.

Architecture mirrors sglang GPU [`mimo_v2_nextn.py`](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/models/mimo_v2_nextn.py):

| Class | Composition |
|---|---|
| `MiMoV2NextNDecoderLayer` | SWA `MiMoV2Attention` (with `attention_sink_bias`) + **dense** `MiMoV2MLP` (the MTP block is *not* MoE, unlike the target's 384-expert layers) + input/post-attn layernorm |
| `MiMoV2ModelNextN` | `embed_tokens` + `enorm`/`hnorm` + `eh_proj` (2h→h) + 1 decoder block + `final_layernorm` |
| `MiMoV2MTPForCausalLM` | `load_lm_head_from_target=True`; `load_weights` handles V2.5-Pro's fused-QKV per-shard FP8 layout via `loader.dequant_fused_qkv` + `dequant_fp8_layers` for the dense MLP |

## Weight mapping

Verified against the live `XiaomiMiMo/MiMo-V2.5-Pro/model_mtp.safetensors` header (48 tensors = 16 per layer × 3 layers):

| HF key (`model.mtp.layers.{idx}.*`) | → JAX target | dtype |
|---|---|---|
| `enorm`/`hnorm`/`final_layernorm`/`input_layernorm` | `model.*` / `model.mtp_block.*` `.scale` | BF16 |
| `pre_mlp_layernorm` | `model.mtp_block.post_attention_layernorm.scale` | BF16 |
| `eh_proj.weight` | `model.eh_proj.weight` (transpose) | BF16 |
| `self_attn.qkv_proj.{weight,weight_scale_inv}` | `__FUSED_QKV_{WEIGHT,SCALE}__0` → split q/k/v via `dequant_fused_qkv` | F8_E4M3 |
| `self_attn.o_proj.weight` | `model.mtp_block.self_attn.o_proj.weight` | BF16 |
| `self_attn.attention_sink_bias` | `model.mtp_block.self_attn.attention_sink_bias` | BF16 |
| `mlp.{gate,up,down}_proj.{weight,weight_scale_inv}` | `model.mtp_block.mlp.*.weight_q`/`.weight_scale` → `dequant_fp8_layers` | F8_E4M3 |

`test_mimo_v2_nextn.py` asserts exact key coverage, per-`mtp_layer_idx` selection (layer N maps only `model.mtp.layers.N.*`), and unique target paths.

## Test plan

- [x] `test_mimo_v2_nextn.py` 3/3 passed (CPU, mapping coverage)
- [x] pre-commit clean
- [ ] E2E load+forward numerical verification — deferred to P1-4 (the `MultiLayerDraftWorker` E2E acceptance "bs=1 accept-len ≥2.5, output 100%=nospec" implies per-layer logits correctness; will fix any `dequant_fused_qkv` interface mismatches there)

### CODE REVIEW

**Comment by zorrofox** [comment]
`performance-test-4-tpu (1)` failure is pre-existing on main — [run 25713412493 on `e2c5f910`](https://github.com/sgl-project/sglang-jax/actions/runs/25713412493) (current main HEAD) fails the exact same check (`test_bench_serving_dense_tp_4.py` throughput threshold 9866). This PR only adds a model class + a mapping unit test and doesn't touch the dense bench path.

File: python/sgl_jax/srt/models/mimo_v2_nextn.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+
+    def set_embed_and_head(self, embed: jax.Array, head: jax.Array) -> None:
+        self.model.embed_tokens.embedding.value = embed
+        self.lm_head.embedding.value = head
+
+    def set_embed(self, embed: jax.Array) -> None:
+        self.model.embed_tokens.embedding.value = embed
+
+
+EntryClass = MiMoV2MTPForCausalLM
```
Comment (pengchengneo) [inline python/sgl_jax/srt/models/mimo_v2_nextn.py:330]: Question / follow-up: this PR adds the MiMoV2 MTP model class, but I don't see the draft architecture selection path wired yet.
 `ModelConfig` currently rewrites draft `MiMoForCausalLM` to `MiMoMTPForCausalLM`, but not `MiMoV2ForCausalLM` to `MiMoV2MTPForCausalLM`.

File: python/sgl_jax/srt/models/mimo_v2_nextn.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        self.loader = WeightLoader(
+            model=self, model_config=model_config, mesh=self.mesh, dtype=self.dtype
+        )
+        mappings = self._create_weight_mappings()
+        self.loader.load_weights_from_safetensors(mappings)
+
+        if self.loader.is_static_quant:
+            head_dim = self.config.swa_head_dim
+            v_head_dim = getattr(self.config, "swa_v_head_dim", head_dim)
+            self.loader.dequant_fused_qkv(
```
Comment (pengchengneo) [inline python/sgl_jax/srt/models/mimo_v2_nextn.py:215]: Question: does this path fully cover the MTP qkv dequant case?

`MiMoV2NextNDecoderLayer` builds attention from the SWA fields  but `dequant_fused_qkv(..., self.config)` still reads the non-SWA fields to infer shards and split Q/K/V.

The fused-QKV FP8 layout looks compatible, but I think the split dims should come from the MTP/SWA attention config, or directly from `mtp_block.self_attn`, to avoid relying on full-attention dims matching MTP dims.

File: python/sgl_jax/srt/models/mimo_v2_nextn.py (hunk 1 above)
Comment (zorrofox) [reply python/sgl_jax/srt/models/mimo_v2_nextn.py:330]: Added the rewrite in `model_config.py` (`MiMoV2ForCausalLM` → `MiMoV2MTPForCausalLM` when `is_draft_model`). `mtp_layer_idx` is still passed via `config.mtp_layer_idx` (default 0); the per-layer override mechanism for the 3-runner case is P1-4's concern. 1be39762.

File: python/sgl_jax/srt/models/mimo_v2_nextn.py (hunk 2 above)
Comment (zorrofox) [reply python/sgl_jax/srt/models/mimo_v2_nextn.py:215]: Correct — `dequant_fused_qkv` reads `config.{head_dim, v_head_dim, num_attention_heads, num_key_value_heads}` (full-attn fields). Changed to derive a `SimpleNamespace` from `mtp_block.self_attn`'s actual dims (`head_dim/v_head_dim/q_head_num/k_head_num`) so the QKV split matches the layer regardless of whether SWA dims equal full-attn dims. Kept the `dequant_fused_qkv` signature unchanged so the V2.5-Pro target call site is unaffected. 1be39762.

**Comment by pengchengneo** [comment]
LGTM

**Review by pengchengneo — APPROVED** [review]
